# Evaluation of state-of-the-art baselines

In [1]:
%load_ext autoreload
%autoreload 2

import evaluation

## Melodia
> Salamon, J., & Gomez, E. (2012). Melody extraction from polyphonic music signals using pitch contour characteristics. IEEE Transactions on Audio, Speech and Language Processing, 20(6), 1759–1770. https://doi.org/10.1109/TASL.2012.2188515

In [2]:
evaluation.summary("salamon", "../algoritmy/essentia-salamon2012-f0-outputs/", ".txt")

Dataset                 ADC04  MDB-mel-s.  MDB-mel-s. valid.   MIREX05  \
Voicing Recall       0.785159    0.794013           0.792896  0.906820   
Voicing False Alarm  0.195785    0.556370           0.434734  0.395940   
Raw Pitch Accuracy   0.675401    0.455051           0.506731  0.756318   
Raw Chroma Accuracy  0.696029    0.533163           0.542397  0.772457   
Overall Accuracy     0.682330    0.438207           0.532277  0.683871   

Dataset              MedleyDB  MedleyDB valid.   ORCHSET    WJazzD  \
Voicing Recall       0.811165         0.846693  0.713354  0.891828   
Voicing False Alarm  0.445023         0.499172  0.482552  0.512337   
Raw Pitch Accuracy   0.500318         0.547118  0.229588  0.686949   
Raw Chroma Accuracy  0.570750         0.607501  0.447482  0.708844   
Overall Accuracy     0.496456         0.515818  0.250601  0.606954   

Dataset              WJazzD valid.  
Voicing Recall            0.898170  
Voicing False Alarm       0.540202  
Raw Pitch Accuracy        0.675119  
Raw Chroma Accuracy       0.710054  
Overall Accuracy          0.584110

## Deep Salience
> Bittner, R. M., Mcfee, B., Salamon, J., Li, P., & Bello, J. P. (2017). Deep Salience Representations for F0 Estimation in Polyphonic Music. Ismir, 23–27.

In [3]:
evaluation.summary("deepsalience", "../algoritmy/ismir2017-deepsalience-f0-outputs/", "_melody2_singlef0.csv")

Dataset                 ADC04  MDB-mel-s.  MDB-mel-s. valid.   MIREX05  \
Voicing Recall       0.796281    0.637848           0.674589  0.795620   
Voicing False Alarm  0.277807    0.273239           0.212752  0.307966   
Raw Pitch Accuracy   0.814213    0.605959           0.757540  0.806656   
Raw Chroma Accuracy  0.854935    0.666314           0.789820  0.823713   
Overall Accuracy     0.715937    0.632738           0.685472  0.701514   

Dataset              MedleyDB  MedleyDB valid.   ORCHSET    WJazzD  \
Voicing Recall       0.674784         0.686728  0.614044  0.845741   
Voicing False Alarm  0.306484         0.248888  0.490377  0.332989   
Raw Pitch Accuracy   0.670104         0.734965  0.518621  0.773577   
Raw Chroma Accuracy  0.735406         0.790595  0.693707  0.785256   
Overall Accuracy     0.610536         0.681659  0.407476  0.691608   

Dataset              WJazzD valid.  
Voicing Recall            0.859023  
Voicing False Alarm       0.357278  
Raw Pitch Accuracy        0.787901  
Raw Chroma Accuracy       0.797715  
Overall Accuracy          0.690570

## Basaran
> D Basaran, S Essid, G. P. (2018). Main melody extraction with source-filter nmf and crnn. Ismir, 82–89.

In [4]:
evaluation.summary("basaran", "../algoritmy/ismir2018_dominant_melody_estimation-f0-outputs/", ".csv")

Dataset                 ADC04  MDB-mel-s.  MDB-mel-s. valid.   MIREX05  \
Voicing Recall       0.731616    0.704252           0.776019  0.713345   
Voicing False Alarm  0.188417    0.270503           0.134364  0.159533   
Raw Pitch Accuracy   0.792810    0.733417           0.860228  0.797663   
Raw Chroma Accuracy  0.819725    0.766182           0.866741  0.807263   
Overall Accuracy     0.668765    0.689269           0.799789  0.734208   

Dataset              MedleyDB  MedleyDB valid.   ORCHSET    WJazzD  \
Voicing Recall       0.676128         0.653918  0.604956  0.841437   
Voicing False Alarm  0.289879         0.246354  0.406912  0.273853   
Raw Pitch Accuracy   0.706123         0.736822  0.634579  0.767087   
Raw Chroma Accuracy  0.757161         0.803256  0.775899  0.775860   
Overall Accuracy     0.640077         0.675413  0.482725  0.700011   

Dataset              WJazzD valid.  
Voicing Recall            0.890464  
Voicing False Alarm       0.360270  
Raw Pitch Accuracy        0.770067  
Raw Chroma Accuracy       0.775065  
Overall Accuracy          0.682325

Ještě výsledky z paperů:
- RPA Durrieu na ORCHSETu = 0.67
- Bosch na ORCHSETu > 0.4

## Bosch - Orchset paper
> Bosch, J. J., Marxer, R., & Gómez, E. (2016). Evaluation and combination of pitch estimation methods for melody extraction in symphonic classical music. Journal of New Music Research, 45(2), 101–117.

Durrieu:

| METRIC | ORCHSET |
| --- | ------- |
| RPA | 0.669   |
| RCA | 0.806   |
| OA  | 0.626   |

In [5]:
evaluation.summary("durrieu", "../algoritmy/durrieu-separateLeadStereo-f0-outputs/", ".txt")

Dataset               ORCHSET
Voicing Recall       0.998633
Voicing False Alarm  0.966964
Raw Pitch Accuracy   0.552295
Raw Chroma Accuracy  0.752982
Overall Accuracy     0.519303